data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,445 @@
+# | ppi_pm                | INTEGER | 1 if found in PPI/PM records, 0 otherwise        |
+
+TABLES_TO_SUPPRESS_QUERY = JINJA_ENV.from_string(
+    """
+    -- Columns for all tables in all datasets --
+    WITH all_columns AS (
+    {% for dataset_id in dataset_ids %}
+     SELECT TABLE_SCHEMA,
+       TABLE_NAME, 
+       COLUMN_NAME,
+       IS_SYSTEM_DEFINED
+     FROM `{{project_id}}.{{dataset_id}}.INFORMATION_SCHEMA.COLUMNS`
+    {% if loop.nextitem is defined %}
+     UNION ALL
+    {% endif %}
+    {% endfor %}
+    ),
+    
+    -- OMOP clinical tables and any tables named <clinical table>_ext --
+    tables AS (
+    SELECT
+      TABLE_SCHEMA as dataset_id,
+      REGEXP_EXTRACT(TABLE_NAME, r"^((?:dose_era|person|cohort_attribute|device_exposure|cohort|procedure_cost|location|observation_period|cohort_definition|drug_exposure|visit_occurrence|note|cost|cdm_source|death|measurement|provider|drug_cost|visit_cost|specimen|condition_occurrence|condition_era|device_cost|attribute_definition|drug_era|observation|payer_plan_period|fact_relationship|care_site|procedure_occurrence)(?:_ext)?)$") AS table_name,
+      ARRAY_AGG(COLUMN_NAME) columns
+    FROM all_columns
+    WHERE 
+      IS_SYSTEM_DEFINED = 'NO'
+      AND table_name IS NOT NULL
+    GROUP BY 1, 2)
+    
+    SELECT 
+     t.dataset_id,
+     -- Name of the OMOP clinical table --
+     t.table_name,
+     -- If it exists, name of corresponding ext table otherwise NULL -- 
+     ext_table.table_name AS ext_table_name,
+     -- Array of column names -- 
+     t.columns
+    FROM tables t
+    
+    -- Corresponding ext table, if any --
+    LEFT JOIN tables ext_table
+      ON  t.dataset_id = ext_table.dataset_id 
+      AND CONCAT(t.table_name, '_ext') = ext_table.table_name
+    WHERE    
+    -- Has a key column --
+    CONCAT(t.table_name, '_id') IN UNNEST(t.columns)
+    
+    -- Has a person_id column --
+    AND 'person_id' IN UNNEST(t.columns)
+    """)
+
+
+def get_tables_to_suppress_df(project_id, target_dataset_ids):
+    """
+    Retrieve table info for datasets to be suppressed
+
+    :param project_id: identifies the project containing the target datasets
+    :param target_dataset_ids: list of dataset_ids of datasets to suppress
+    :return: dataframe with dataset_id, table_name, ext_table_name, columns
+    """
+    query = TABLES_TO_SUPPRESS_QUERY.render(project_id=project_id, dataset_ids=target_dataset_ids)
+    query_job = CLIENT.query(query)
+    tables_to_wipe_df = query_job.to_dataframe()
+    return tables_to_wipe_df
+
+
+ROWS_TO_SUPPRESS_QUERY = JINJA_ENV.from_string("""
+SELECT DISTINCT
+ '{{dataset_id}}'                         
+                                          AS dataset_id,
+ '{{table}}'
+                                          AS table,
+ t.{{table}}_id
+                                          AS row_id,
+ t.person_id                              AS person_id,
```

Comment:
nit:  why is there so much whitespace between the selection and the alias?

Reply:
This weird formatting is me trying to ease query comprehension, both in the template and also in template renderings. I find that visually aligning the aliases makes it easier to call out the field names (i.e. in the result set). The alignment would be thrown off in rendered templates because the length of associated expressions (left hand side of 'AS') have varying lengths, so I put the alias on a separate line.

I realize these are idiosyncratic choices- formatting would ideally ease comprehension for **all** team members. I will keep this for now for fear that I will have to read this code again soon, but how about we plan to discuss standard ways to make our SQL more understandable? I will bring this up at our next sync meeting.

Reply:
ok.  we should discuss.  sounds like a plan.